odl/solvers/nonsmooth/admm.py

```diff
@@ -159,3 +170,212 @@ def admm_linearized_simple(x, f, g, L, tau, sigma, niter, **kwargs):
         u = L(x) + u - z
         if callback is not None:
             callback(x)
+
+
+def admm_precon_nonlinear(x, f, g, L, delta, niter, sigma=None, **kwargs):
```

Comment:
Not sure that "precon" should have its own function, IMO it should just be optional and this can be called "admm_nonlinear"

Reply:
Also, are we sure that this is *the* admm_nonlinear method, or can we get a naming collision here in the future?

Reply:
As I understand it, the preconditioning is part of the method, and the way it is used makes the method explicit, i.e., everything can be expressed in terms of proximals. *The* ADMM method has implicit steps in the form of some `argmin` which is not a proximal.